src/Sarif/UriHelper.cs

```diff
@@ -69,6 +71,16 @@ public static string MakeValidUri(string path)
                     !path.StartsWith(@"\", StringComparison.Ordinal))
                 {
                     validUri = validUri.Substring(1);
+
+                    // When the UriBuilder constructs an absolute URI, it strips any
+                    // leading "." and ".." segments ("dot-segments", as RFC 3986 calls
+                    // them). Glue them back on so we don't lose the relative path
+                    // information.
+                    string leadingDotSegments = GetLeadingDotSegments(path);
+                    if (!string.IsNullOrEmpty(leadingDotSegments))
+                    {
+                        validUri = leadingDotSegments + validUri;
+                    }
```

Comment:
Interesting. I'm surprised because there is a test for `Relative Additional\Location` becoming `Relative%20Adittional/Location`. I would expect that test to fail and output `/Relative%20Additional/Location`

A quick interactive test does show that I am broken for the ../ case, though.

https://github.com/dotnet/roslyn/pull/11390/commits/c66413016d28dde42a1cd6cc669b3c4d47835deb